TREnvironmentEditor/Model/Types/Triggers/EMCameraTriggerFunction.cs

```diff
@@ -1,228 +1,225 @@
-                    AttachToSector(sector, control, cameraIndex, data);
-                }
+                TR2Entity attachToEntity = level.Entities[data.ConvertEntity(item)];
+                TRRoomSector sector = FDUtilities.GetRoomSector(attachToEntity.X, attachToEntity.Y, attachToEntity.Z, data.ConvertRoom(attachToEntity.Room), level, control);
+                AttachToSector(sector, control, cameraIndex, data);
             }
+        }
 
-            if (AttachToLocations != null)
+        if (AttachToLocations != null)
+        {
+            foreach (EMLocation location in AttachToLocations)
             {
-                foreach (EMLocation location in AttachToLocations)
-                {
-                    TRRoomSector sector = FDUtilities.GetRoomSector(location.X, location.Y, location.Z, data.ConvertRoom(location.Room), level, control);
-                    AttachToSector(sector, control, cameraIndex, data);
-                }
+                TRRoomSector sector = FDUtilities.GetRoomSector(location.X, location.Y, location.Z, data.ConvertRoom(location.Room), level, control);
+                AttachToSector(sector, control, cameraIndex, data);
             }
+        }
 
-            if (AttachToRooms != null)
+        if (AttachToRooms != null)
+        {
+            foreach (short room in AttachToRooms)
             {
-                foreach (short room in AttachToRooms)
+                foreach (TRRoomSector sector in level.Rooms[data.ConvertRoom(room)].Sectors)
                 {
-                    foreach (TRRoomSector sector in level.Rooms[data.ConvertRoom(room)].Sectors)
-                    {
-                        AttachToSector(sector, control, cameraIndex, data);
-                    }
+                    AttachToSector(sector, control, cameraIndex, data);
                 }
             }
-
-            control.WriteToLevel(level);
         }
 
-        private TRCamera CreateCamera(EMLevelData data)
+        control.WriteToLevel(level);
+    }
+
+    private TRCamera CreateCamera(EMLevelData data)
+    {
+        return new TRCamera
         {
-            return new TRCamera
-            {
-                X = Camera.X,
-                Y = Camera.Y,
-                Z = Camera.Z,
-                Room = data.ConvertRoom(Camera.Room),
-                Flag = Camera.Flag,
-            };
-        }
+            X = Camera.X,
+            Y = Camera.Y,
+            Z = Camera.Z,
+            Room = data.ConvertRoom(Camera.Room),
+            Flag = Camera.Flag,
+        };
+    }
 
-        private void AttachToSector(TRRoomSector sector, FDControl control, ushort cameraIndex, EMLevelData data)
+    private void AttachToSector(TRRoomSector sector, FDControl control, ushort cameraIndex, EMLevelData data)
+    {
+        if (sector.FDIndex != 0)
         {
-            if (sector.FDIndex != 0)
+            if (control.Entries[sector.FDIndex].Find(e => e is FDTriggerEntry) is FDTriggerEntry trigger
```

Comment:
I don't know about you, but this variant is less readable to me than the original; if it's normal in C# world or the IDE complains that it should be coded this way, let's leave it as-is.